src/node_file.h

```diff
@@ -21,10 +21,43 @@ using v8::Value;
 
 namespace fs {
 
+// structure used to store state during a complex operation, e.g., mkdirp.
+class FSContinuationData : public MemoryRetainer {
+ public:
+  FSContinuationData(uv_fs_t* req, int mode, uv_fs_cb done_cb)
+      : req(req), mode(mode), done_cb(done_cb) {
+  }
+
+  uv_fs_t* req;
+  int mode;
+  std::vector<std::string> paths;
+
+  void PushPath(const std::string& path) {
+    paths.push_back(path);
+  }
+  std::string PopPath() {
+    CHECK_GE(paths.size(), 0);
+    std::string path = paths[paths.size() - 1];
```

Comment:
Similarly, this also creates a copy but doesn’t really need to – I think `path = std::move(paths.back())` works?